### PR TITLE
chore(opt8n): Refactor `opt8n` into `cmd` modules

### PR DESCRIPTION
**Pull request by 0xKitsune**
This PR refactors `opt8n` into `cmd` models, similar to `opd8n`.

```rust

#[derive(Parser, Clone, Debug)]
pub enum Commands {
    Repl(ReplArgs),
    Script(ScriptArgs),
    Server(ServerArgs),
}

#[tokio::main]
async fn main() -> eyre::Result<()> {
    color_eyre::install()?;
    let command = Args::parse().command;

    match command {
        Commands::Repl(cmd) => cmd.run().await?,
        Commands::Script(cmd) => cmd.run().await?,
        Commands::Server(cmd) => cmd.run().await?,
    }

    Ok(())
}

```

### CODE REVIEW

File: bin/opt8n/src/cmd/server.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+impl ServerArgs {
+    pub async fn run(&self) -> color_eyre::Result<()> {
+        // let opt8n = Opt8n::new(
+        //     Some(self.node_args.clone()),
+        //     self.opt8n_args.output.clone(),
+        //     self.opt8n_args.genesis.clone(),
+        // )
+        // .await?;
+
+        Ok(())
```
Comment (refcell) [inline bin/opt8n/src/cmd/server.rs:23]: Preference to replace this with the [`unimplemented` macro](https://doc.rust-lang.org/std/macro.unimplemented.html).

```suggestion
        unimplemented!()
```

File: examples/exec-scripts/script/Counter.s.sol [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (refcell) [inline examples/exec-scripts/script/Counter.s.sol:1]: Did we want to commit this example?

**Review by refcell — APPROVED** [review]
Looks good, just a few nits/questions

File: examples/exec-scripts/script/Counter.s.sol (hunk 2 above)
Comment (0xKitsune) [reply examples/exec-scripts/script/Counter.s.sol]: Good catch, I just removed this.

**Review by refcell — APPROVED** [review]

**Pull request merged**